src/Microsoft.Azure.WebJobs.Host/Diagnostics/ExceptionFormatter.cs

```diff
@@ -0,0 +1,228 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Adapted from code here: https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioErrorReportingService.ExceptionFormatting.cs
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Host.Diagnostics
+{
+    /// <summary>
+    /// Provides functionality to format and sanitize an exception for logging.
+    /// </summary>
+    public static class ExceptionFormatter
+    {
+        private static readonly string[] EmtpyArray = new string[0];
+
+        /// <summary>
+        /// Formats an exception for logging.
+        /// </summary>
+        /// <param name="exception">The exception to be formatted.</param>
+        /// <returns>A string representation of the exception with a sanitized stack trace.</returns>
+        public static string GetFormattedException(Exception exception)
+        {
+            var aggregate = exception as AggregateException;
+            if (aggregate != null)
+            {
+                return GetStackForAggregateException(exception, aggregate);
+            }
+
+            return GetStackForException(exception, includeMessageOnly: false);
```

Comment:
In all the grovelling through stack frames, etc. accessing members, etc. I'm wondering if in strange cases (e.g. generated code, etc.) what the possibilities are for some info to not be there (e.g. null refs) causing things to blow up. Wondering if we shouldn't have a try/catch surrounding all this that defaults to the full stack trace if for some reason our sanitization logic blows up?